File: app/src/main/java/com/itis/android/lessontwo/di/module/AppModule.java
package com.itis.android.lessontwo.di.module;

import android.content.Context;

import com.itis.android.lessontwo.App;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Nail Shaykhraziev on 09.04.2018.
 */
@Module
public class AppModule {

    private App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Provides
    @Singleton
    Context provideApp() {
        return app;
    }
}

File: app/src/main/java/com/itis/android/lessontwo/di/module/AppModule.kt
package com.itis.android.lessontwo.di.module

import android.content.Context
import com.itis.android.lessontwo.App
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Singleton
    @Provides
    fun bindContext(application: App): Context = application.applicationContext
}
